eclass/kernel-build.eclass

```diff
@@ -382,6 +382,21 @@ kernel-build_src_install() {
 	local module_ver
 	module_ver=$(<"${relfile}") || die
 
+	# warn when trying to "make" a dist-kernel
+	cat <<-EOF >> "${ED}${kernel_dir}/Makefile" || die
+		_IS_USER_SHELL:=\$(shell [ -t 0 ] && echo 1)
```

Comment:
Could you add `_GENTOO` to it, just in case? And an empty line above it, so it doesn't glue to whatever was above.

Reply:
Good idea, done!